Crash in table manager on unexpected data

```
ts=2018-04-20T11:11:14.262762309Z caller=log.go:112 level=info msg="checking provisioned throughput on table" table=prod_chunk_data_weekly_2519
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xae944e]

goroutine 69 [running]:
github.com/weaveworks/cortex/pkg/chunk/aws.dynamoTableClient.DescribeTable.func4.1(0x170da40, 0xc420871050, 0xc420871050, 0x170da40)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/aws/dynamodb_table_client.go:238 +0x31e
github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument.CollectedRequest(0x170da40, 0xc420212630, 0xf1370b, 0x39, 0x170a700, 0xc42036e2d8, 0xf21c30, 0xc4205433d0, 0xe0dba0, 0xc420543300)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument/instrument.go:143 +0x1dd
github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument.TimeRequestHistogram(0x7f04a40d8000, 0xc420212630, 0xf1370b, 0x39, 0xc42018c390, 0xc4205433d0, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument/instrument.go:169 +0xd6
github.com/weaveworks/cortex/pkg/chunk/aws.dynamoTableClient.DescribeTable.func4(0x170da40, 0xc420212630, 0xc420212630, 0x0)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/aws/dynamodb_table_client.go:227 +0xef
github.com/weaveworks/cortex/pkg/chunk/aws.dynamoTableClient.backoffAndRetry(0x171aee0, 0xc42018c480, 0x1715020, 0xc42018c4a0, 0xc420322690, 0x5f5e100, 0xba43b7400, 0x14, 0x170da40, 0xc420212630, ...)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/aws/dynamodb_table_client.go:78 +0x133
github.com/weaveworks/cortex/pkg/chunk/aws.dynamoTableClient.DescribeTable(0x171aee0, 0xc42018c480, 0x1715020, 0xc42018c4a0, 0xc420322690, 0x5f5e100, 0xba43b7400, 0x14, 0x170da40, 0xc420212630, ...)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/aws/dynamodb_table_client.go:226 +0x3ca
github.com/weaveworks/cortex/pkg/chunk/aws.(*dynamoTableClient).DescribeTable(0xc42038f940, 0x170da40, 0xc420212630, 0xc420218fc0, 0x1b, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:1 +0x100
github.com/weaveworks/cortex/pkg/chunk.(*TableManager).updateTables(0xc42034b8c0, 0x170da40, 0xc420212630, 0xc42055a000, 0x73, 0x9d, 0x0, 0x0)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/table_manager.go:273 +0x2a2
github.com/weaveworks/cortex/pkg/chunk.(*TableManager).SyncTables(0xc42034b8c0, 0x170da40, 0xc420212630, 0x84df200, 0x170da40)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/table_manager.go:175 +0x370
github.com/weaveworks/cortex/pkg/chunk.(*TableManager).loop.func1(0x170da40, 0xc420212630, 0xc420212630, 0x170da40)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/table_manager.go:141 +0x3e
github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument.CollectedRequest(0x170d9c0, 0xc42001a018, 0xef8374, 0x17, 0x170a700, 0xc42022a028, 0xf21c30, 0xc42030df38, 0xe0db00, 0xc420220600)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument/instrument.go:143 +0x1dd
github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument.TimeRequestHistogram(0x7f04a5923410, 0xc42001a018, 0xef8374, 0x17, 0xc42018c350, 0xc420220738, 0xc42006c000, 0xc4202b9eb0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/common/instrument/instrument.go:169 +0xd6
github.com/weaveworks/cortex/pkg/chunk.(*TableManager).loop(0xc42034b8c0)
	/go/src/github.com/weaveworks/cortex/pkg/chunk/table_manager.go:140 +0x147
created by github.com/weaveworks/cortex/pkg/chunk.(*TableManager).Start
	/go/src/github.com/weaveworks/cortex/pkg/chunk/table_manager.go:125 +0x5f
```

--- a/pkg/chunk/aws/dynamodb_table_client.go
+++ b/pkg/chunk/aws/dynamodb_table_client.go
@@ -139,7 +139,9 @@ func (d dynamoTableClient) CreateTable(ctx context.Context, desc chunk.TableDesc
 			if err != nil {
 				return err
 			}
-			tableARN = output.TableDescription.TableArn
+			if output.TableDescription != nil {
+				tableARN = output.TableDescription.TableArn
+			}
 			return nil
 		})
 	}); err != nil {
@@ -175,11 +177,24 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 			out, err := d.DynamoDB.DescribeTableWithContext(ctx, &dynamodb.DescribeTableInput{
 				TableName: aws.String(name),
 			})
+			if err != nil {
+				return err
+			}
 			desc.Name = name
-			desc.ProvisionedRead = *out.Table.ProvisionedThroughput.ReadCapacityUnits
-			desc.ProvisionedWrite = *out.Table.ProvisionedThroughput.WriteCapacityUnits
-			status = *out.Table.TableStatus
-			tableARN = out.Table.TableArn
+			if out.Table != nil {
+				if provision := out.Table.ProvisionedThroughput; provision != nil {
+					if provision.ReadCapacityUnits != nil {
+						desc.ProvisionedRead = *provision.ReadCapacityUnits
+					}
+					if provision.WriteCapacityUnits != nil {
+						desc.ProvisionedWrite = *provision.WriteCapacityUnits
+					}
+				}
+				if out.Table.TableStatus != nil {
+					status = *out.Table.TableStatus
+				}
+				tableARN = out.Table.TableArn
+			}
 			return err
 		})
 	})
@@ -192,6 +207,9 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 			out, err := d.DynamoDB.ListTagsOfResourceWithContext(ctx, &dynamodb.ListTagsOfResourceInput{
 				ResourceArn: tableARN,
 			})
+			if err != nil {
+				return err
+			}
 			desc.Tags = make(map[string]string, len(out.Tags))
 			for _, tag := range out.Tags {
 				desc.Tags[*tag.Key] = *tag.Value
@@ -208,14 +226,25 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 					ScalableDimension: aws.String("dynamodb:table:WriteCapacityUnits"),
 					ServiceNamespace:  aws.String("dynamodb"),
 				})
+				if err != nil {
+					return err
+				}
 				switch l := len(out.ScalableTargets); l {
 				case 0:
 					return err
 				case 1:
 					desc.WriteScale.Enabled = true
-					desc.WriteScale.RoleARN = *out.ScalableTargets[0].RoleARN
-					desc.WriteScale.MinCapacity = *out.ScalableTargets[0].MinCapacity
-					desc.WriteScale.MaxCapacity = *out.ScalableTargets[0].MaxCapacity
+					if target := out.ScalableTargets[0]; target != nil {
+						if target.RoleARN != nil {
+							desc.WriteScale.RoleARN = *target.RoleARN
+						}
+						if target.MinCapacity != nil {
+							desc.WriteScale.MinCapacity = *target.MinCapacity
+						}
+						if target.MaxCapacity != nil {
+							desc.WriteScale.MaxCapacity = *target.MaxCapacity
+						}
+					}
 					return err
 				default:
 					return fmt.Errorf("more than one scalable target found for DynamoDB table")
@@ -231,13 +260,25 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 					ScalableDimension: aws.String("dynamodb:table:WriteCapacityUnits"),
 					ServiceNamespace:  aws.String("dynamodb"),
 				})
+				if err != nil {
+					return err
+				}
 				switch l := len(out.ScalingPolicies); l {
 				case 0:
 					return err
 				case 1:
-					desc.WriteScale.InCooldown = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.ScaleInCooldown
-					desc.WriteScale.OutCooldown = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.ScaleOutCooldown
-					desc.WriteScale.TargetValue = *out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration.TargetValue
+					config := out.ScalingPolicies[0].TargetTrackingScalingPolicyConfiguration
+					if config != nil {
+						if config.ScaleInCooldown != nil {
+							desc.WriteScale.InCooldown = *config.ScaleInCooldown
+						}
+						if config.ScaleOutCooldown != nil {
+							desc.WriteScale.OutCooldown = *config.ScaleOutCooldown
+						}
+						if config.TargetValue != nil {
+							desc.WriteScale.TargetValue = *config.TargetValue
+						}
+					}
 					return err
 				default:
 					return fmt.Errorf("more than one scaling policy found for DynamoDB table")
@@ -292,7 +333,9 @@ func (d dynamoTableClient) UpdateTable(ctx context.Context, current, expected ch
 				if err != nil {
 					return err
 				}
-				tableARN = out.Table.TableArn
+				if out.Table != nil {
+					tableARN = out.Table.TableArn
+				}
 				return nil
 			})
 		}); err != nil {

